### ISSUE TITLE
v2.1.0 seems not compatible with Uglify (JS minification)

### ISSUE THREAD

**Issue by masonlouchart** [vaadin/vaadin-control-state-mixin#43]
I'm using `vaadin-text-field` in my application and this element has `vaadin-control-state-mixin` as dependency. I updated my dependencies today and the app stopped to compile. After investigation I fixed the version of `vaadin-control-state-mixin` to _v2.0.3_ and the application compiles again.

**Am I alone to get this issue?**

Here a part of my build pipe:

```javascript
const uglifyjs = require('uglify-es');
const composer = require('gulp-uglify/composer');
const uglify = composer(uglifyjs, console);

// ...

let dependenciesStream = polymerProject.dependencies()
      .pipe(dependenciesStreamSplitter.split())
      // Optimizations.
      .pipe(gulpif(/\.js$/, uglify()))
      .pipe(gulpif(/\.css$/, cssSlam({
        stripWhitespace: true,
      })))
      .pipe(gulpif(/\.html$/, htmlMinifier({
        removeComments: true,
        collapseWhitespace: true,
        removeTagWhitespace: true,
        minifyCSS: true,
      })))
      .pipe(dependenciesStreamSplitter.rejoin());
```

It fails on the `uglify()` call. The error is related to the `vaadin-control-state-mixin.html_script0.js` (that is generated on the fly by the stream splitter I guess) and the message is _"Cannot read property indexOf of undefined"_. My theory is the stream splitter encounters difficulty to extract the JavaScript from the HTML file. The syntax seems good. I'l try to figure it out by checking the differences between the 2 versions. Feel free to help on it.

**Comment by masonlouchart** [comment]
I revert locally the commit [179c6d4](https://github.com/vaadin/vaadin-control-state-mixin/commit/179c6d44ae9dbdb9185d23286be9d759d5c62a24) and build my app successfully. Then I revert the revert to return to the v2.1.1, try again and get the same error as previously.

So I conclude that removing the `TabIndex` from the Vaadin namespace is the origin of my issue. I don't know why.

**Comment by web-padawan** [comment]
Thanks for the issue. I will try to reproduce but I need more info about versions, especially of `uglify-es` which you use.

**Comment by masonlouchart** [comment]
I'm using [gulp-uglify](https://github.com/terinjokes/gulp-uglify) v3.0.0 and [uglify-es](https://github.com/mishoo/UglifyJS2/tree/harmony) v3.3.2.

**Comment by web-padawan** [comment]
@masonlouchart so I have tried the versions you mentioned and has not been able to reproduce the issue. Here is the repo which you could fork and confirm it works for with `uglify-es` 3.3.9:

1. Clone the repo
```sh
git clone https://github.com/web-padawan/gulp-minify-test && cd gulp-minify-test
```
2. Install dependencies
```sh
npm i && bower i
```
3. Run minification
```sh
npm start
```
4. Check the minified content:
```
dist/bower_components/vaadin-control-state-mixin/vaadin-control-state-mixin.html
```

When downgrading to 3.3.2 it does throw an error though.

**Comment by web-padawan** [comment]
Appears to be addressed by mishoo/UglifyJS2#2771 which fixed processing some JS comments. 
Since minifying works with latest version, I'm closing this issue. The code in the mixin is totally valid JavaScript and thus we are not inclined to apply any changes to it.

**Comment by masonlouchart** [comment]
Thank you for your help @web-padawan!

_I guess I didn't upgrade properly my modules the first time.
Like you said, the syntax is correct that's why I was confused._
